packages/react-test-renderer/src/ReactShallowRenderer.js

```diff
@@ -172,12 +183,21 @@ function basicStateReducer<S>(state: S, action: BasicStateAction<S>): S {
   return typeof action === 'function' ? action(state) : action;
 }
 
+type ShallowRenderOptions = {
+  callEffects: boolean,
+};
+
+const isEffectHook = Symbol();
```

Comment:
Ultra paranoia to prevent any other non effect hook from creating a memoized state that could look like an effect hook's memoized state.